src/test/java/com/connexta/search/index/IndexTests.java

```diff
@@ -6,286 +6,139 @@
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.multipart;
-import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.mockito.Mockito.verifyNoInteractions;
 
+import com.connexta.search.common.exceptions.DetailedErrorAttributes;
+import com.connexta.search.index.configs.IndexControllerConfiguration;
 import com.connexta.search.index.controllers.IndexController;
-import com.connexta.search.index.exceptions.IndexException;
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
+import com.connexta.search.index.controllers.IndexControllerUnitTests;
+import com.connexta.search.rest.models.IndexRequest;
+import java.net.URI;
 import javax.inject.Inject;
-import javax.validation.constraints.NotEmpty;
-import javax.validation.constraints.NotNull;
-import org.apache.commons.io.IOUtils;
-import org.apache.solr.client.solrj.SolrClient;
-import org.apache.solr.client.solrj.SolrRequest.METHOD;
-import org.apache.solr.client.solrj.SolrServerException;
-import org.apache.solr.client.solrj.response.QueryResponse;
-import org.apache.solr.common.SolrDocumentList;
-import org.apache.solr.common.SolrInputDocument;
-import org.apache.solr.common.StringUtils;
-import org.apache.solr.common.params.SolrParams;
-import org.geotools.data.DataStore;
 import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.ValueSource;
-import org.mockito.Mock;
-import org.mockito.junit.jupiter.MockitoExtension;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.data.repository.CrudRepository;
-import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
-import org.springframework.mock.web.MockMultipartFile;
-import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.reactive.server.WebTestClient;
 
-@ExtendWith(MockitoExtension.class)
-@SpringBootTest
-@AutoConfigureMockMvc
-@MockBean(DataStore.class)
+/**
+ * This class contains tests that use {@link
+ * org.springframework.test.web.reactive.server.WebTestClient} to test the {@link IndexController}
+ * that cannot be tested in {@link IndexControllerUnitTests} and that do not interact with the
+ * {@link IndexService}.
+ */
+@WebFluxTest({
+  IndexController.class,
+  IndexControllerConfiguration.class,
+  DetailedErrorAttributes.class
+})
 public class IndexTests {
```

Comment:
✏️ Our naming in tests is inconsistent.

Reply:
See https://github.com/connexta/ion-search/pull/66#discussion_r342156459

Reply:
See https://github.com/connexta/ion-search/pull/66#discussion_r342156459